lua/luasnip/extras/expand_conditions.lua

```diff
@@ -1,13 +1,62 @@
 local M = {}
 
-function M.line_begin(line_to_cursor, matched_trigger)
+-----------------------
+-- CONDITION OBJECTS --
+-----------------------
+local condition_mt = {
+	-- logic operators
+	-- not '~'
+	__unm = function(o1)
+		return function(...)
+			return not o1(...)
+		end
+	end,
+	-- or '+'
+	__add = function(o1, o2)
+		return function(...)
+			return o1(...) or o2(...)
+		end
+	end,
+	-- and '*'
+	__mul = function(o1, o2)
+		return function(...)
+			return o1(...) and o2(...)
+		end
+	end,
+	-- xor '^'
+	__pow = function(o1, o2)
+		return function(...)
+			return o1(...) ~= o2(...)
+		end
+	end,
+	-- xnor '=='
+	__eq = function(o1, o2)
+		return function(...)
+			return o1(...) == o2(...)
+		end
+	end,
+	-- use table like a function by overloading __call
+	__call = function(tab, line_to_cursor, matched_trigger, captures)
+		return tab.func(line_to_cursor, matched_trigger, captures)
+	end,
+}
+function M.condition_factory(func)
```

Comment:
WDYT of `make_condition` instead? Feels a bit more descriptive

Reply:
Yes, `factory` is comprehensive if you've been working with something like this before, but `make_condition` should be better in general.